### PR TITLE
Update gen unit tests to JUnit

### PR DESCRIPTION
**Pull request by Damans227**
Linked to #9378

- Build was successful after making the changes
- All tests passed locally in the IDE

List of classes updated: 

`core/trino-main/src/test/java/io/trino/sql/gen/TestInCodeGenerator.java`
`core/trino-main/src/test/java/io/trino/sql/gen/TestJoinCompiler.java`
`core/trino-main/src/test/java/io/trino/sql/gen/TestPageFunctionCompiler.java`

### CODE REVIEW

**Comment by ebyhr** [comment]
> Build was successful after making the changes

What command did you use? I'm asking this question because checkstyle is failing:
```
2021-09-29T00:53:17.8756945Z [INFO] There are 3 errors reported by Checkstyle 8.41.1 with checkstyle/airbase-checks.xml ruleset.
2021-09-29T00:53:17.8780955Z [ERROR] src/test/java/io/trino/sql/gen/TestJoinCompiler.java:[237] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
2021-09-29T00:53:17.8783364Z [ERROR] src/test/java/io/trino/sql/gen/TestJoinCompiler.java:[250] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
2021-09-29T00:53:17.8785317Z [ERROR] src/test/java/io/trino/sql/gen/TestJoinCompiler.java:[257] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
```

**Comment by Damans227** [comment]
> > Build was successful after making the changes
> 
> What command did you use? I'm asking this question because checkstyle is failing:
> ```
> 2021-09-29T00:53:17.8756945Z [INFO] There are 3 errors reported by Checkstyle 8.41.1 with checkstyle/airbase-checks.xml ruleset.
> 2021-09-29T00:53:17.8780955Z [ERROR] src/test/java/io/trino/sql/gen/TestJoinCompiler.java:[237] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
> 2021-09-29T00:53:17.8783364Z [ERROR] src/test/java/io/trino/sql/gen/TestJoinCompiler.java:[250] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
> 2021-09-29T00:53:17.8785317Z [ERROR] src/test/java/io/trino/sql/gen/TestJoinCompiler.java:[257] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
> ```

Hmm, that's odd. 

Here is the command I used: 

`./mvnw -pl '!:trino-server-rpm,!:trino-docs,!:trino-proxy,!:trino-verifier,!:trino-benchto-benchmarks' clean install \
-TC2 -nsu \
-DskipTests \
-Dmaven.javadoc.skip=true \
-Dmaven.source.skip=true \
-Dair.check.skip-all=true`

I found the command in this blog: 

https://trino.io/episodes/20.html

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java [hunk 1]
```diff
@@ -170,8 +174,10 @@
 import static java.util.Collections.emptyMap;
 import static java.util.Collections.nCopies;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
 
-@Test(singleThreaded = true)
+@TestInstance(PER_CLASS)
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java:179]: How to make it default for the project?

**Comment by hashhar** [comment]
@Damans227 The "quick build" command is useful if you just want to test that things compile. It skips all linters like dependency checks, checkstyle etc.

`./mvnw clean install -TC4 -DskipTests` is what you can use to check everything (since that's what CI uses).

**Review by findepi — COMMENTED** [review]
Same comment as in https://github.com/trinodb/trino/pull/9388#discussion_r716513806
this mustn't be merged before that is resolved.

**Comment by Damans227** [comment]
Thanks for sharing that command @hashhar I will revisit the changes in this PR using the new command.

**Comment by Damans227** [comment]
Closing this and other PRs related to JUnit migration for now.

**Pull request closed without merging**